cvat/apps/iam/views.py

```diff
@@ -215,10 +217,17 @@ def get(self, request):
 class OAuth2CallbackViewEx(OAuth2CallbackView):
     def dispatch(self, request, *args, **kwargs):
         # Distinguish cancel from error
-        if (auth_error := request.GET.get('error', None)) and \
-            auth_error == self.adapter.login_cancelled_error:
-            return HttpResponseRedirect(settings.SOCIALACCOUNT_CALLBACK_CANCELLED_URL)
-        return super().dispatch(request, *args, **kwargs)
+        if (auth_error := request.GET.get('error', None)):
+            if auth_error == self.adapter.login_cancelled_error:
+                return HttpResponseRedirect(settings.SOCIALACCOUNT_CALLBACK_CANCELLED_URL)
+            else: # unknown error
+                raise ValidationError(auth_error)
+
+        code = request.GET.get('code')
+        state = request.GET.get('state')
+        assert code and state, 'Parameters code and state not found in request'
```

Comment:
It should not an assertion. Our users can use our API incorrectly. Need to return BadRequest error with corresponding error.